source/slang/slang-serialize-container.h

```diff
@@ -0,0 +1,107 @@
+// slang-serialize-container.h
+#ifndef SLANG_SERIALIZE_CONTAINER_H
+#define SLANG_SERIALIZE_CONTAINER_H
+
+#include "../core/slang-riff.h"
+#include "slang-serialize-types.h"
+#include "slang-ir-insts.h"
+#include "slang-profile.h"
+
+namespace Slang {
+
+class EndToEndCompileRequest;
+
+/* The binary representation actually held in riff/file format*/
+struct SerialContainerBinary
+{
+    struct Target
+    {
+        uint32_t target;
+        uint32_t flags;
+        uint32_t profile;
+        uint32_t floatingPointMode;
+    };
+
+    struct EntryPoint
+    {
+        uint32_t name;
+        uint32_t profile;
+        uint32_t mangledName;
+    };
+};
+
+/* Struct that holds all the data that can be held in a 'container' */
+struct SerialContainerData
+{
+    struct Target
+    {
+        CodeGenTarget       codeGenTarget = CodeGenTarget::Unknown;
+        SlangTargetFlags    flags = 0;
+        Profile             profile;
+        FloatingPointMode   floatingPointMode = FloatingPointMode::Default;
+    };
+    
+    struct TargetModule
+    {
+        // IR module for a specific compilation target
+        Target target;
+        RefPtr<IRModule> irModule;
+    };
+
+    struct TranslationUnit
+    {
+        RefPtr<IRModule> irModule;
+        RefPtr<ASTBuilder> astBuilder;
+        NodeBase* astRootNode = nullptr;
+    };
```

Comment:
Similarly, this `TranslationUnit` type is precisely what the front-end logic in `slang-compiler.h` calls a `Module`, and it is *not* the same as a `TranslationUnitRequest`.